### PR TITLE
build: build binary tarball from static binaries.

### PR DESCRIPTION
**Pull request by klihub**
Add targets for building static binaries and cleaning up built binaries. Change `binary-dist` to build tarball from static binaries.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -241,13 +241,16 @@ all: build update-workflows
 
 build: $(BUILD_BINS)
 
+build-static: clean-bin
```
Comment (marquiz) [inline Makefile:244]: Should we add the `clean-bin` dep to `build:` instead? Just thought about altering static and non-static builds...

Comment (klihub) [reply Makefile:244]: Hmm, yes. Maybe that's a better idea.

Comment (klihub) [reply Makefile]: Okay, I updated it so, that binaries are rebuilt if
1. they are out of date, or
2. existing binaries were built with a different STATIC setting than the current one

**Review by askervin — APPROVED** [review]
LGTM

**Review by marquiz — APPROVED** [review]
lgtm

**Pull request merged**